scripts/cocoapods/__tests__/utils-test.rb

```diff
@@ -382,6 +382,44 @@ def test_fixReactBridgingHeaderSearchPaths_correctlySetsTheHeaderSearchPathsForA
         end
     end
 
+    # ===================================== #
+    # Test - Apply Xcode14 React-Core patch #
+    # ===================================== #
+
+    def test_turnOffResourceBundleReactCore_correctlyAppliesPatch
+        # Arrange
+        react_core_debug_config = prepare_Code_Signing_YES_build_configuration("Debug")
+        react_core_release_config = prepare_Code_Signing_YES_build_configuration("Release")
+
+        user_project_mock = UserProjectMock.new("a/path", [
+            prepare_config("Debug"),
+            prepare_config("Release"),
+        ])
+
+        pods_projects_mock = PodsProjectMock.new([],
+            {
+            "hermes-engine" => {},
+            "React-Core" => [ react_core_debug_config, react_core_release_config ],
+        })
+
+        installer = InstallerMock.new(pods_projects_mock, [
+            AggregatedProjectMock.new(user_project_mock)
+        ])
+
```

Comment:
I would probably add and assert similar to the one we have at line 416 to make sure that the `CODE_SIGNING_ALLOWED` flag is YES before the invocation of `turn_off_resource_bundle_react_core`. In this way the test properly assert that we update from `YES` to `NO`.